autoload/go/path.vim

```diff
@@ -138,14 +138,16 @@ endfunction
 function! go#path#CheckBinPath(binpath)
     " remove whitespaces if user applied something like 'goimports   '
     let binpath = substitute(a:binpath, '^\s*\(.\{-}\)\s*$', '\1', '')
+    " save off original path
+    let old_path = $PATH
 
     " check if we have an appropriate bin_path
     let go_bin_path = go#path#BinPath()
-
-    " append our GOBIN and GOPATH paths and be sure they can be found there...
-    " let us search in our GOBIN and GOPATH paths
-    let old_path = $PATH
-    let $PATH = go_bin_path . go#util#PathListSep() . $PATH
+    if empty(go_bin_path)
+        " append our GOBIN and GOPATH paths and be sure they can be found there...
+        " let us search in our GOBIN and GOPATH paths
+        let $PATH = go_bin_path . go#util#PathListSep() . $PATH
+    endif
```

Comment:
Ok I didn't quite get this. If `go_bin_path` is empty, this means we're going to prepend an empty string to PATH. Why do we prepend an empty string ? We shouldn't do that :)